plugins/terminal.koplugin/main.lua

```diff
@@ -83,6 +83,17 @@ local T = require("ffi/util").template
 
 local CHUNK_SIZE = 80 * 40 -- max. nb of read bytes (reduce this, if taps are not detected)
 
+-- This should be moved inside `platform/android/luajit-launcher/assets/android.lua`
+-- once we now it works as expected
+if Device:isAndroid() and not os.getenv("SHELL") and os.execute() == nil then
+    local dummy, android = pcall(require, "android")
+    os.execute = function(command) -- luacheck: ignore 122
+        command = command .. "; exit $?"
```

Comment:
The exit code from the shell is already the code from the last command.

Reply:
In which case is this not true?

Reply:
> The exit code from the shell is already the code from the last command.

Yup.

> The equivalent of a bare exit is exit $? or even just omitting the exit.

(c.f., https://tldp.org/LDP/abs/html/exit-status.html#AEN2981)

Granted, this is Bash, but I assume this is the standard POSIX behavior. Anything else would be utter madness.

Reply:
Android doesn't care about standard POSIX behavior. Hard to imagine it wouldn't do the exit status in a semi-normal manner though.

Reply:
https://android.googlesource.com/platform/system/core/+/master/shell_and_utilities/README.md says

> Since IceCreamSandwich Android has used [mksh](https://www.mirbsd.org/mksh.htm) as its shell. Before then it used [ash](https://en.wikipedia.org/wiki/Almquist_shell) (which actually remained unused in the tree up to and including KitKat).

Hard to imagine a Korn shell from a BSD distro fucking up that bad either ;).